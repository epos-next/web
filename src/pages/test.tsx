import React, { useState } from "react";
import styled from "styled-components";
import ModalWindowBase from "@components/modal-window-base";

export default function Test() {
    const [open, setOpen] = useState(false);

    return <Page>
        <button onClick={ () => setOpen(true) }>open</button>
        <ModalWindowBase onClose={ () => null } isOpen={ open }>
            <h2>Some content</h2>
            <button onClick={ () => setOpen(false) }>close</button>
        </ModalWindowBase>
    </Page>
}


const Page = styled.main`
`;
