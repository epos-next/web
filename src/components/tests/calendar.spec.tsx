import React from "react";
import CalendarComponent, { getDates } from "@components/calendar";
import DateHelper from "@helpers/date-helper";
import { fireEvent, render } from "@testing-library/react";
import "jest-styled-components"
import { getDatesAnswers } from "../../../test/fixtures";

describe("testing calendar component", () => {
    const now = new Date(2021, 11, 21)

    beforeEach(() => {
        jest.spyOn(DateHelper, "now", "get").mockReturnValue(now)
    })

    afterAll(() => {
        jest.clearAllMocks()
    })

    it("should match snapshot", () => {
        const { asFragment } = render(<CalendarComponent/>)
        expect(asFragment()).toMatchSnapshot()
    });

    it("should handle day change", () => {
        const callback = jest.fn()
        const { getByTestId } = render(<CalendarComponent onDayChanged={ callback }/>)
        const date = new Date(2021, 11, 11)
        fireEvent.click(getByTestId(`date-${ date.toISOString() }`))
        expect(callback).toBeCalledWith(date)
    })

    it("should call callback with today date on initialize", () => {
        const callback = jest.fn()
        render(<CalendarComponent onDayChanged={ callback }/>)
        expect(callback).toBeCalledTimes(1)
        expect(callback).toBeCalledWith(now)
    });

    it("should change month on if user select date with another month", () => {
        const callback = jest.fn()
        const { getByTestId } = render(<CalendarComponent onDayChanged={ callback }/>)
        const date = new Date(2021, 10, 30)
        fireEvent.click(getByTestId(`date-${ date.toISOString() }`))
        expect(callback).toBeCalledWith(date)
        expect(getByTestId("month")).toHaveProperty("value", "2")
    });

    it("should change month on on user select", () => {
        const callback = jest.fn()
        const { getByTestId } = render(<CalendarComponent onDayChanged={ callback }/>)
        fireEvent.change(getByTestId("month"), { target: { value: 3 } })
        expect(callback).toBeCalledWith(new Date(2021, 11, 21))
        expect(getByTestId("month")).toHaveProperty("value", "3")
    });

    describe("testing getDates()", () => {
        it("should return correct dates array (2021-12-21)", () => {
            const date = new Date(2021, 11, 21)
            const answer = getDatesAnswers[0]
            const received = getDates(date.getFullYear(), date.getMonth())
            expect(received).toEqual(answer)
            expect(received.length % 7).toEqual(0)
        })

        it("should return correct dates array (2021-02-09)", () => {
            const date = new Date(2021, 1, 9)
            const answer = getDatesAnswers[1]
            const received = getDates(date.getFullYear(), date.getMonth())
            expect(received).toEqual(answer)
            expect(received.length % 7).toEqual(0)
        })

        it("should return correct dates array (2021-05-17)", () => {
            const date = new Date(2021, 4, 17)
            const answer = getDatesAnswers[2]
            const received = getDates(date.getFullYear(), date.getMonth())
            expect(received).toEqual(answer)
            expect(received.length % 7).toEqual(0)
        })

        it("should return correct dates array (2021-01-01)", () => {
            const date = new Date(2021, 1, 1)
            const answer = getDatesAnswers[3]
            const received = getDates(date.getFullYear(), date.getMonth())
            expect(received).toEqual(answer)
            expect(received.length % 7).toEqual(0)
        })
    })
})
