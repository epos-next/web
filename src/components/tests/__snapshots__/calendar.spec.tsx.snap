// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing calendar component should match snapshot 1`] = `
<DocumentFragment>
  .c8 {
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  color: white;
}

.c8:after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  margin: 0 auto;
  border-radius: 50%;
  background: linear-gradient(328.45deg,#656DFD 11.68%,#949EFD 85.36%);
  width: 36px;
  height: 36px;
  z-index: -1;
}

.c7 {
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--primary);
  cursor: pointer;
}

.c6 {
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--light-primary);
  cursor: pointer;
}

.c4 {
  display: grid;
  grid-template-columns: repeat(7,1fr);
  margin-top: 5px;
}

.c5 {
  font-size: 15px;
  line-height: 18px;
  color: var(--secondary);
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  font-weight: 500;
  font-size: 18px;
  line-height: 22px;
  color: var(--primary);
}

.c3 {
  font-size: 18px;
  line-height: 22px;
  color: var(--primary);
  -webkit-appearance: none;
  -moz-appearance: none;
  text-indent: 1px;
  text-overflow: '';
  outline: none;
  text-align: right;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 calc((100% / 7 - 18px) / 2);
}

.c0 {
  width: 282px;
}

@media screen and (max-width:960px) {
  .c0 {
    width: calc(100%);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <h5
        class="c2"
      >
        Календарь
      </h5>
      <select
        class="c3"
        data-testid="month"
        dir="rtl"
      >
        <option
          value="0"
        >
          Сентябрь
        </option>
        <option
          value="1"
        >
          Октябрь
        </option>
        <option
          value="2"
        >
          Ноябрь
        </option>
        <option
          value="3"
        >
          Декабрь
        </option>
        <option
          value="4"
        >
          Январь
        </option>
        <option
          value="5"
        >
          Февраль
        </option>
        <option
          value="6"
        >
          Март
        </option>
        <option
          value="7"
        >
          Апрель
        </option>
        <option
          value="8"
        >
          Май
        </option>
        <option
          value="9"
        >
          Июнь
        </option>
        <option
          value="10"
        >
          Июль
        </option>
        <option
          value="11"
        >
          Август
        </option>
      </select>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        Пн
      </div>
      <div
        class="c5"
      >
        Вт
      </div>
      <div
        class="c5"
      >
        Ср
      </div>
      <div
        class="c5"
      >
        Чт
      </div>
      <div
        class="c5"
      >
        Пт
      </div>
      <div
        class="c5"
      >
        Сб
      </div>
      <div
        class="c5"
      >
        Вс
      </div>
      <div
        class="c6"
        data-testid="date-2021-11-28T19:00:00.000Z"
      >
        29
      </div>
      <div
        class="c6"
        data-testid="date-2021-11-29T19:00:00.000Z"
      >
        30
      </div>
      <div
        class="c7"
        data-testid="date-2021-11-30T19:00:00.000Z"
      >
        1
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-01T19:00:00.000Z"
      >
        2
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-02T19:00:00.000Z"
      >
        3
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-03T19:00:00.000Z"
      >
        4
      </div>
      <div
        class="c6"
        data-testid="date-2021-12-04T19:00:00.000Z"
      >
        5
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-05T19:00:00.000Z"
      >
        6
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-06T19:00:00.000Z"
      >
        7
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-07T19:00:00.000Z"
      >
        8
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-08T19:00:00.000Z"
      >
        9
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-09T19:00:00.000Z"
      >
        10
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-10T19:00:00.000Z"
      >
        11
      </div>
      <div
        class="c6"
        data-testid="date-2021-12-11T19:00:00.000Z"
      >
        12
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-12T19:00:00.000Z"
      >
        13
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-13T19:00:00.000Z"
      >
        14
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-14T19:00:00.000Z"
      >
        15
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-15T19:00:00.000Z"
      >
        16
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-16T19:00:00.000Z"
      >
        17
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-17T19:00:00.000Z"
      >
        18
      </div>
      <div
        class="c6"
        data-testid="date-2021-12-18T19:00:00.000Z"
      >
        19
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-19T19:00:00.000Z"
      >
        20
      </div>
      <div
        class="c8"
        data-testid="date-2021-12-20T19:00:00.000Z"
      >
        21
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-21T19:00:00.000Z"
      >
        22
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-22T19:00:00.000Z"
      >
        23
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-23T19:00:00.000Z"
      >
        24
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-24T19:00:00.000Z"
      >
        25
      </div>
      <div
        class="c6"
        data-testid="date-2021-12-25T19:00:00.000Z"
      >
        26
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-26T19:00:00.000Z"
      >
        27
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-27T19:00:00.000Z"
      >
        28
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-28T19:00:00.000Z"
      >
        29
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-29T19:00:00.000Z"
      >
        30
      </div>
      <div
        class="c7"
        data-testid="date-2021-12-30T19:00:00.000Z"
      >
        31
      </div>
      <div
        class="c6"
        data-testid="date-2021-12-31T19:00:00.000Z"
      >
        1
      </div>
      <div
        class="c6"
        data-testid="date-2022-01-01T19:00:00.000Z"
      >
        2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
